src/test/java/bowlingexercise3/BowlingTest.java

```diff
@@ -44,16 +39,40 @@ public void testBowling10AndMore(String turns, int score) {
     @ParameterizedTest
     @CsvSource(value = {
 
-            //Spare tests
+            //Spare tests frames 1-9
             "-/, 10",
+            "-/|-/, 20",
+            "-/|-/|-/, 30",
+            "-/|-/|-/|-/, 40",
             "-/|43|-8|25|53|-4|45|-7|-/, 74",
             "-/|43|-/|-/|-/|-/|45|-/|-/, 94",
             "-/|9/|-/|-/|5/|-/|45|-/|-/, 107",
             "-/|9/|-/|-/|5/|-7|45|9/|-/, 100",
             "-/|-/|-/|-/|-/|-/|-/|7/|-/, 97",
             "1/|-/|4/|9/|, 53"})
 
-    public void testBowlingspares(String turns, int score) {
+    public void testBowlingSpares(String turns, int score) {
+        Bowling game = new Bowling();
+        game.turns(turns);
+        assertEquals(score, game.score());
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+
+            //Strike tests frames 1-9
+            "X, 10",
+            "X|11|, 14",
+            "X|5/|1-, 32",
+            "X|X|X|22, 70",
+            "X|43|-8|X|53|-4|45|-7|X, 88",
+            "X|43|X|X|X|X|45|X|X, 166",
+            "X|9/|X|X|52|X|45|X|X, 147",
+            "-/|17|X|-/|5/|X|X|9/|X, 153",
+            "-/|X|X|-/|-/|X|-/|7/|-/, 147",
+            "X|X|X|X|X|X|X|X|X, 240"})
+
+    public void testBowlingStrikes(String turns, int score) {
```

Comment:
A good set of tests made me think about the scoring! In this case, it's identifying all the different cases for the next two balls after the strike that could occur which is quite comprehensive! Maybe another of the initial cases might be a strike followed by a miss with just two frames as that covers all the cases with a small data set. That way you test all the strike cases before adding in the mix of spares and other scores across a larger number of frames?

Reply:
Have now included this test